patch/savefile.txt

```diff
@@ -0,0 +1,202 @@
+PatchSavefile:
+	call CheckNotEmpty
+	call CheckSavefileIsValid
+	call PrintSavefileMessage
+	; set #address, VersionInfo
+	; getbyteinc #result, #address
+	; getbyte #temp, #address
+	; mulacum #temp, #result, 0x100
+	set #temp, CURRENT_BUILD
+	ifgt #build, 9999, .invalid_build
+	ifgt #build, #temp, .future_build
+	ifeq #build, #temp, .latest_build
+	push #temp
+	bufstring .savefile_build_string_1
+	set #1, #build
+	set #2, 4
+	call PrintWithLeadingZeros
+	bufstring .savefile_build_string_2
+	printbuf
+	menu #result, YesNoMenu
+	jumpnz #result, .refused
+	; call PokecenterCheck
+	set #address, SavePatches
+.loop
+	gethalfwordinc #curbuild, #address
+	bufchar 0x76
+	bufnumber #curbuild
+	printbuf
+	jumpz #curbuild, .done
+	getwordinc #patch, #address
+	ifle #curbuild, #build, .loop
+	push #address
+	call #patch
+	pop #address
+	jump .loop
+.done
+	pop #build
+	set #curbuild, #build
+	call FixSavefileChecksum
+	; seek sBuildNumber
+	; writehalfword #build
```

Comment:
You're doing this elsewhere, so you should delete this

Reply:
This is actually slightly different from the one I'm using in save_patches_old.asm as that's an initial way to set save patch versions that FIQ had set up, it will likely be different after newbox.